src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -0,0 +1,402 @@
+    if (strongDelegate == nil || delegateQueue == nil) {
+        LogErrorOnFailure(CHIP_ERROR_INCORRECT_STATE);
+        AsyncResponder::NotifyEventHandled(event, CHIP_ERROR_INCORRECT_STATE);
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    dispatch_async(delegateQueue, ^{
+        if ([strongDelegate respondsToSelector:@selector(handleBDXQueryForNodeID:
+                                                                      controller:blockSize:blockIndex:bytesToSkip:completion:)]) {
+            [strongDelegate handleBDXQueryForNodeID:nodeId
+                                         controller:controller
+                                          blockSize:blockSize
+                                         blockIndex:blockIndex
+                                        bytesToSkip:bytesToSkip
+                                         completion:completionHandler];
+        } else {
+            [strongDelegate handleBDXQueryForNodeID:nodeId
+                                         controller:controller
+                                          blockSize:blockSize
+                                         blockIndex:blockIndex
+                                        bytesToSkip:bytesToSkip
+                                  completionHandler:completionHandler];
+        }
+    });
+    return CHIP_NO_ERROR;
+}
+
+void MTROTAImageTransferHandler::HandleTransferSessionOutput(TransferSession::OutputEvent & event)
+{
+    VerifyOrReturn(mDelegate != nil);
+
+    ChipLogError(BDX, "OutputEvent type: %s", event.ToString(event.EventType));
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    switch (event.EventType) {
+    case TransferSession::OutputEventType::kInitReceived:
+        err = OnTransferSessionBegin(event);
+        if (err != CHIP_NO_ERROR) {
+            LogErrorOnFailure(err);
+            AsyncResponder::NotifyEventHandled(event, err);
+        }
+        break;
+    case TransferSession::OutputEventType::kStatusReceived:
+        ChipLogError(BDX, "Got StatusReport %x", static_cast<uint16_t>(event.statusData.statusCode));
+        [[fallthrough]];
+    case TransferSession::OutputEventType::kAckEOFReceived:
+    case TransferSession::OutputEventType::kInternalError:
+    case TransferSession::OutputEventType::kTransferTimeout:
+        err = OnTransferSessionEnd(event);
+        if (err != CHIP_NO_ERROR) {
+            LogErrorOnFailure(err);
+            AsyncResponder::NotifyEventHandled(event, err);
+        }
+        break;
+    case TransferSession::OutputEventType::kQueryWithSkipReceived:
+    case TransferSession::OutputEventType::kQueryReceived:
+        err = OnBlockQuery(event);
+        if (err != CHIP_NO_ERROR) {
+            LogErrorOnFailure(err);
+            AsyncResponder::NotifyEventHandled(event, err);
+        }
+        break;
+    case TransferSession::OutputEventType::kNone:
+    case TransferSession::OutputEventType::kAckReceived:
+        // Nothing to do.
+        break;
```

Comment:
so this is the OTA provider side so we do not handle the messages but when the OTA requestor uses the AsyncTransferFacilitator it will handle these messages. it would send acks for the blocks it receives.

Reply:
move all messages not handled to chipdie

Reply:
done.